mcp/mcp-server/src/main/java/software/amazon/smithy/java/mcp/server/McpServer.java

```diff
@@ -113,12 +120,29 @@ private void handleRequest(JsonRpcRequest req) {
                         InitializeResult.builder()
                                 .capabilities(Capabilities.builder()
                                         .tools(Tools.builder().listChanged(true).build())
+                                        .prompts(Prompts.builder().listChanged(true).build())
                                         .build())
                                 .serverInfo(ServerInfo.builder()
                                         .name(name)
                                         .version("1.0.0")
                                         .build())
                                 .build());
+                case "prompts/list" -> writeResponse(req.getId(),
+                        ListPromptsResult.builder().prompts(prompts.values().stream().toList()).build());
```

Comment:
We need to prevent the prompt templates from being serialized in this response. We only want the name, title, description, and arguments.